### DISCUSSION TITLE
GH Pages endless "DNS Check in Progress"

### DISCUSSION THREAD

**Discussion by ShaneCurcuru** [community/community#37200] [Other Feature Feedback, Questions, & Ideas]
I had a custom domain that worked over http:, even though it kept giving the DNS check messages on the repo's Pages page.  I tried turning on enforce https:, but that never really worked since the Pages still had the message, and it broke the site.

I turned off Pages for over a week, and just turned it back on, after triple checking my DNS records. Same error.

Is this repo just borked on the GitHub internals side somehow? Or did I indeed mess up my DNS records? I'm hoping to not have to fork into a new repo name to get Pages working. I'm suspicious, because this happened once with another throwaway site: one repo got stuck on the DNS check, and would never correct itself. dig shows the domain has the correct A records.

https://github.com/ShaneCurcuru/slides/commit/d34041a6a73ef3f84161a5c19a4d45133dcc0be5

**Comment by shellscape** [comment]
I think this is just another post-Microsoft rough edge in a sea of rough edges that will likely never get fixed. I have several sites with valid HTTPS in Custom Domains that are stuck on `DNS Check in Progress`.

**Comment by tanlucvn** [comment]
Response #2

**Comment by Tanyike** [comment]
Does your server has a valid ssl certificate ?

**Reply by shellscape** [reply]
my dude GitHub issues the SSL certs. what are you talking about?

**Comment by peterroth** [comment]
My page also has the "DNS Check in Progress" marker, however I can access it with https, what is using a valid, signed certificate.

**Comment by ajdamico** [comment]
i'm also experiencing this issue, with a domain purchased through google workstation and hosted by enom.com ..

**Comment by pplmx** [comment]
Same issue. Endless checking. It's been working fine for the past few years, but in the last few weeks it's been acting up.

**Comment by jfullstackdev** [comment]
maybe Copilot suggestions can help

---

It's hard to say for sure without more information, but here are a few things you can check:

1. **DNS Propagation**: DNS changes can take up to 48 hours to fully propagate. If you've recently made changes to your DNS records, it's possible that GitHub is still seeing the old records. You can use a tool like [DNS Checker](https://dnschecker.org/) to check the propagation status of your DNS records.

2. **DNS Configuration**: Make sure your DNS records are configured correctly according to [GitHub's documentation](https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/managing-a-custom-domain-for-your-github-pages-site#configuring-an-apex-domain). For an apex domain, you should have 4 A records pointing to the following IP addresses:
   - 185.199.108.153
   - 185.199.109.153
   - 185.199.110.153
   - 185.199.111.153

   If you're using a www subdomain, you should have a CNAME record pointing to your GitHub Pages domain (`username.github.io`).

3. **HTTPS Enforcement**: If you've enabled HTTPS enforcement in your repository settings, make sure your custom domain supports HTTPS. Some domain providers do not support HTTPS for apex domains.

4. **GitHub Pages Settings**: Check your repository settings to make sure the custom domain is correctly set. The custom domain should be entered in the "Custom domain" field in the GitHub Pages section of your repository settings.

If you've checked all of these things and are still having issues, it might be worth reaching out to GitHub Support for assistance. They can check the status of your Pages site from their end and may be able to provide more insight into what's going wrong.
